[blog] GitLab support (#279)
gitpod-io/website: src/blog/gitlab-support.md

---
date: Wed Dec 04 2019 13:49:24 GMT+0000 (UTC)
author: svenefftinge
subtitle: Your devops toolchain is finally complete!
title: GitLab Support for Gitpod is Here 🎉
image: /teaser-gitlab.gif
---

We are happy to announce that you can finally enjoy Gitpod's frictionless ready-to-code dev environments on GitLab as well.

![gitpod loves gitlab](./gitlab-support/teaser-gitlab-gitpod.jpg)

# Getting Started with GitLab and Gitpod

Starting a dev environment for a GitLab project works just as with GitHub, i.e. you prefix any GitLab project URL with `gitpod.io/#`.

> You can try it out with the Spring Petclinic example:

Reviewer: ```suggestion
You can try it out with the Spring PetClinic example:
```

 https://gitpod.io/#https://gitlab.com/gitpod/spring-petclinic

Besides the standard project URL, Gitpod supports
 - file context - [Example](https://gitpod.io/#https://gitlab.com/gitpod/spring-petclinic/blob/master/src/main/java/org/springframework/samples/petclinic/PetClinicApplication.java)
 - branch context - [Example](https://gitpod.io/#https://gitlab.com/gitpod/spring-petclinic/tree/my-branch),
 - issue context - [Example](https://gitpod.io/#https://gitlab.com/gitpod/spring-petclinic/issues/1) and

>  - merge request context - [Example](https://gitpod.io/#https://gitlab.com/gitpod/spring-petclinic/merge_requests/1)

Reviewer: Maybe it's just me, but the word "context" seems a bit "alien" to me (i.e. GitHub or GitLab rarely use the word "context", so I have to guess what it means here with respect to Gitpod/GitHub/GitLab).

If you share my feeling, this could be worked around e.g. like so:

> Besides opening simple project root URLs, Gitpod also supports:
> - opening files - example
> - opening branches - example
> - opening issues - example
> - opening merge requests - example

Reviewer: Ah, maybe the sentence below is enough to explain what this means. Feel free to ignore my suggestion.

Depending on the context, your dev environment will be initialized differently.

> For instance, if you start a workspace from the issue context above, you’ll get a fresh workspace with a local branch, branched-off from `master`.

Reviewer: ```suggestion
For instance, if you start a workspace from the issue context above, you’ll get a fresh workspace with a new local branch based on `master`.
```

The new local branch will be named like `<username>/<issue-title>-<issue-nr>`, in my case it is:

> `svenefftinge/allow-skip-first-name-1` (The number of included words from the issue’s title is limited, of course, depending on their length).

Reviewer: Maybe clarify like so?

```suggestion
`svenefftinge/allow-skip-first-name-1` (if the issue title is too long, only the first few words will be included in the branch name).
```

## Configure your project

> To get the most out of Gitpod, you need to tell it how a perfect dev environment looks like. Dev environments run as containers which you can configure through custom Docker files.

Reviewer 2: ```suggestion
To get the most out of Gitpod, you need to tell it how a perfect dev environment for your project looks like. Dev environments run as containers which you can configure through custom Docker files.
```

> To get the most out of Gitpod, you need to tell it how a perfect dev environment looks like. Dev environments run as containers which you can configure through custom Docker files.

Reviewer: Optional suggestion, because "perfect dev environment" sounds strange to me:

```suggestion
To get the most out of Gitpod, you need to tell it what your dev environment should look like. Dev environments run as containers which you can configure through custom Docker files.
```

> Further more you can specify which tasks should run after a fresh checkout. Usually this includes build steps, downloading dependencies and running some unit tests. Read below's section on prebuilds, to make Gitpod automatically run these steps as a CI pipeline whenever someone pushes changes to your project.

Reviewer: ```suggestion
Furthermore you can specify which tasks should run after a fresh checkout. Usually this includes build steps, downloading dependencies and running some unit tests. Read below's section on prebuilds, to make Gitpod automatically run these steps as a CI pipeline whenever someone pushes changes to your project.
```

There is much more you can do to make sure every team member gets a ready-to-code dev environment when they need one. Please refer to the [docs](/docs/40_configuration/) for details.

## Prebuilds

> Gitpod’s [prebuilt workspaces](/docs/46_prebuilds/) are key to providing ephemeral, ready-to-code dev environments for your branches and projects. You can configure Gitpod to run your project's build asynchronously every time someone pushes new commits or branches to your repository. So when a developer wants to start coding, she can, because everything is already prepared.

Reviewer: Suggestion: the point of prebuilt workspaces is not to make workspaces _ephemeral_, but to make them _near-instant_

```suggestion
Gitpod’s [prebuilt workspaces](/docs/46_prebuilds/) are key to providing near-instant, ready-to-code dev environments for your branches and projects. You can configure Gitpod to run your project's build asynchronously every time someone pushes new commits or branches to your repository. So when a developer wants to start coding, she can, because everything is already prepared.
```

> Enabling prebuilds on your GitLab project is done by configuring a corresponding Webhook. But don't fear, you don't have to do that manually, just run a prebuild on your project and it will be configured automatically. To do that, use the prefix `https://gitpod.io/#prebuild/`, as in

Reviewer: The word "corresponding" doesn't add information but makes the sentence sound more complicated.

```suggestion
Enabling prebuilds on your GitLab project is done by configuring a Webhook. But don't fear, you don't have to do that manually, just run a prebuild on your project and it will be configured automatically. To do that, use the prefix `https://gitpod.io/#prebuild/`, as in
```

Reviewer: Also, in GitLab, I only need to visit `gitpod.io/#prebuild/<repo>` once, and it will be prebuilt automatically forever? That was a bit surprising to me.
